pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -247,19 +267,27 @@ func (r *Builder) Secret(vmRef ref.Ref, in, object *core.Secret) (err error) {
 	if err != nil {
 		return
 	}
+	thumbprint := string(in.Data["thumbprint"])
 	if hostDef, found := r.hosts[hostID]; found {
 		hostSecret, nErr := r.hostSecret(hostDef)
 		if nErr != nil {
 			err = nErr
 			return
 		}
 		in = hostSecret
+		// Get host thumbprint
+		h, nErr := r.host(hostID)
+		if nErr != nil {
+			err = nErr
+			return
+		}
+		thumbprint = h.Thumbprint
```

Comment:
yeah, I think it's better the way you did it compared to set it on the secret here (in.Data["thumbprint"] = h.Thumbprint)

Reply:
It's better not to change the source provider secret. It should always contain the VCenter thumbprint.

Reply:
yeah, I meant without persisting the changes into etcd, just so we'll operate on the secret below - but I do like the way you wrote it more